### PR TITLE
Make Docker workflow configurable

### PR DESCRIPTION
**Pull request by dhillondeep**
- Add ability to run individual services
- Add documentation regarding workflow

cc @uwblueprint/the-shoe-project

### CODE REVIEW

**Comment by dhillondeep** [comment]
Should be merged after #59

File: docker-compose.yml
Comment (AbhiPrasad) [inline docker-compose.yml]: `uwblueprint.org` is more correct here for us, but I guess it doesn't really matter.

File: docker-compose.yml
Comment (AbhiPrasad) [inline docker-compose.yml:46]: Why run both `yarn start` and `yarn watch`?

Isn't this duplicating how we serve assets? Don't really see a need for node running it's own server.

If we decide to move this way though, it might make sense for us to server side render with a small express server. That way we can hydrate and make first page load way faster (we don't have to do it rn, we can do it whenever, even next term). See: https://reactjs.org/docs/react-dom-server.html

Comment (AbhiPrasad) [reply docker-compose.yml:46]: +1 on moving it to another service tho

File: docker/Dockerfile.dev
Comment (AbhiPrasad) [inline docker/Dockerfile.dev]: let's go with exec form if it's a single command, it has better optimizations + works better with signals iirc.

```suggestion
CMD ["go", "run", "./shoe-project", "server"]
```

**Review by AbhiPrasad — COMMENTED** [review]
Will approve after other PR is merged in.

File: docker-compose.yml
Comment (dhillondeep) [reply docker-compose.yml]: yeah i did not realize it was `org` lol will fix

File: docker-compose.yml
Comment (dhillondeep) [reply docker-compose.yml:46]: I made `frontend` run its own node server because this service can run independently and frontend developers don't need go server to run. When we are running all services, we can just ignore that node server

File: docker/Dockerfile.dev
Comment (dhillondeep) [reply docker/Dockerfile.dev]: right! looks much cleaner

**Review by AbhiPrasad — APPROVED** [review]

**Pull request merged**
